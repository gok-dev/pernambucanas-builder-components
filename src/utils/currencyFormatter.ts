export const currencyFormatter = (amount: number) => {
  return parseFloat(amount.toString()).toLocaleString('pt-BR', {
    minimumFractionDigits: 2,
    style: 'currency',
    currency: 'BRL'
  })
}
