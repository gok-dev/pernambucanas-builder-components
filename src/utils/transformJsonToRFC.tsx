import { Banner } from 'index'
import * as React from 'react'

const rFCStringMapper = {
  Banner: Banner
}

export const transformJsonToRFC = (
  json: { name: string; props: any }[],
  onChange: (param: any) => void
): React.ReactElement[] => {
  const Components = json?.map(item => {
    return React.createElement(rFCStringMapper[item.name], {
      ...item.props,
      onChange
    })
  })

  return Components
}
