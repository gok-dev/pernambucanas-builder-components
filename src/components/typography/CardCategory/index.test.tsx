import React from 'react'
import { renderWithTheme } from 'utils/renderWithTheme'
import { CardCategory } from './index'

describe( '<CardCategory />', () => {
  it( 'shold render the heading', () => {
    const { container } = renderWithTheme(
      <CardCategory title="Sacola de desconto" />
    )

    expect( container.firstChild ).toMatchSnapshot()
  } )
} )
