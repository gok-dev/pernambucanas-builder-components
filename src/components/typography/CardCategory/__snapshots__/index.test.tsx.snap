// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCategory /> shold render the heading 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  background-color: #FFFFFF;
  border-radius: 8px;
  width: 100%;
  min-height: 100px;
  padding: 18px;
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0:hover {
  cursor: pointer;
  box-shadow: 2px 2px 10px rgba(0,0,0,0.17);
}

<article
  class="c0"
  radius="8"
>
  <div
    class="c1"
  />
</article>
`;
